lib/response.js

```diff
@@ -766,13 +766,29 @@ internals.Response.Peek = class extends Stream.Transform {
         super();
 
         this._podium = podium;
-        this.on('finish', () => podium.emit('finish'));
     }
 
-    _transform(chunk, encoding, callback) {
+    async _transform(chunk, encoding, callback) {
 
-        this._podium.emit('peek', [chunk, encoding]);
-        this.push(chunk, encoding);
-        callback();
+        try {
+            await this._podium.emit('peek', [chunk, encoding]);
+
+            callback(null, chunk);
```

Comment:
Missing the encoding argument.

Reply:
This callback() call should be outside the try-catch. Otherwise we risk calling it twice (if it somehow throws).

Reply:
[callback](https://nodejs.org/api/stream.html#stream_transform_transform_chunk_encoding_callback) does not seem to take a third argument (for `encoding`).

Event Node's [PassThrough](https://github.com/nodejs/node/blob/9c62e0e384e2ea5a45a4011f31c97be60f2a0737/lib/internal/streams/passthrough.js#L45-L47) just calls `callback(null, chunk)`

Do you want me to revert to the following anyway ?

```js
        this.push(chunk, encoding);
        callback();
```

Reply:
Well, i just did revert to the way it was. Let's not change what's working, am i right ?!

Reply:
Great